public/app/plugins/datasource/grafana-azure-monitor-datasource/datasource.ts

```diff
@@ -84,28 +68,28 @@ export default class Datasource extends DataSourceApi<AzureMonitorQuery, AzureDa
       }
 
       // Initialize the list of queries
-      let q = byType[target.queryType];
-      if (!q) {
-        q = _.cloneDeep(options);
-        q.requestId = `${q.requestId}-${target.refId}`;
-        q.targets = [];
-        byType[target.queryType] = q;
+      if (!byType.has(target.queryType)) {
+        const queryForType = _.cloneDeep(options);
+        queryForType.requestId = `${queryForType.requestId}-${target.refId}`;
+        queryForType.targets = [];
+        byType.set(target.queryType, queryForType);
       }
-      q.targets.push(target);
+
+      const queryForType = byType.get(target.queryType);
+      queryForType && queryForType.targets.push(target);
```

Comment:
nit: can be simplified:
```suggestion
      queryForType?.targets.push(target);
```